src-tauri/src/database/repository.rs

```diff
@@ -0,0 +1,939 @@
+        encounter_id: i64,
+    ) -> Result<()> {
+
+        let InsertEncounterArgs {
+            encounter,
+            cast_log,
+            damage_log,
+            skill_cast_log,
+            player_info,
+            party_info,
+            ..
+        } = args;
+
+        let fight_start = encounter.fight_start;
+        let fight_end = encounter.last_combat_packet;
+
+        let buffs = compute_support_buffs(encounter, party_info);
+
+        let mut statement = transaction.prepare_cached(INSERT_ENTITY)?;
+
+        for (_name, entity) in encounter.entities.iter() {
+            if !should_insert_entity(entity, &encounter.local_player) {
+                continue;
+            }
+
+            let mut entity = entity.clone();
+
+            update_entity_stats(&mut entity, fight_start, fight_end, damage_log);
+
+            if let Some(info) = player_info.as_ref().and_then(|stats| stats.get(&entity.name)) {
+                apply_player_info(&mut entity, info);
+            }
+
+            apply_cast_logs(&mut entity, cast_log, skill_cast_log);
+
+            let compressed_skills = compress_json(&entity.skills)?;
+            let compressed_damage_stats = compress_json(&entity.damage_stats)?;
+
+            let support = buffs.get(&entity.name);
+
+            let params = params![
+                entity.name,
+                encounter_id,
+                entity.npc_id,
+                entity.entity_type.to_string(),
+                entity.class_id,
+                entity.class,
+                entity.gear_score,
+                entity.current_hp,
+                entity.max_hp,
+                entity.is_dead,
+                compressed_skills,
+                compressed_damage_stats,
+                json!(entity.skill_stats),
+                entity.damage_stats.dps,
+                entity.character_id,
+                json!(entity.engraving_data),
+                entity.loadout_hash,
+                entity.combat_power,
+                entity.ark_passive_active,
+                entity.spec,
+                json!(entity.ark_passive_data),
+                support.map(|b| b.buff),
+                support.map(|b| b.brand),
+                support.map(|b| b.identity),
+                support.map(|b| b.hyper),
+                entity.damage_stats.unbuffed_damage,
+                entity.damage_stats.unbuffed_dps
+            ];
+
+            statement.execute(params)?;
+        }
+
+        Ok(())
+    }
+
+    fn insert_encounter_preview(
+        &self,
+        transaction: &Transaction,
+        args: &InsertEncounterArgs,
+        encounter_id: i64,
+    ) -> Result<()> {
+
+        let InsertEncounterArgs {
+            encounter,
+            raid_clear,
+            raid_difficulty,
+            ..
+        } = args;
+
+        let mut players: Vec<_> = encounter.entities.values()
+            .filter(|e| ((e.entity_type == EntityType::Player && e.class_id != 0 && e.max_hp > 0)
```

Comment:
```suggestion
            .filter(|e| ((e.entity_type == EntityType::Player && e.class_id > 0)
```
from https://github.com/snoww/loa-logs/commit/16bea4394c61d80a61457cae27477e431adf6d60 to match https://github.com/snoww/loa-logs/blob/88c9cd4211728b90ce188672e87b970c178e5dbe/src-tauri/src/database/utils.rs#L384